packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -147,8 +149,9 @@ export default function parse(source: string, state: State): TemplateParseResult
     const stack: IRElement[] = [];
 
     const preserveComments =
```

Comment:
The same way that `preserveComments` is collected prior to the AST traversal, it would also make sense to collected render mode here.